source-jdbc/src/main/java/org/apache/kylin/source/jdbc/JdbcHiveInputBase.java

```diff
@@ -229,4 +299,232 @@ protected static String escapeQuotationInSql(String sqlExpr) {
+
+        StringBuilder whereBuilder = new StringBuilder();
+        whereBuilder.append("WHERE 1=1");
+
+        DataModelDesc model = flatDesc.getDataModel();
+        if (StringUtils.isNotEmpty(model.getFilterCondition())) {
+            whereBuilder.append(" AND (").append(model.getFilterCondition()).append(") ");
+        }
+
+        if (flatDesc.getSegment() != null) {
+            PartitionDesc partDesc = model.getPartitionDesc();
+            if (partDesc != null && partDesc.getPartitionDateColumn() != null) {
+                SegmentRange segRange = flatDesc.getSegRange();
+
+                if (segRange != null && !segRange.isInfinite()) {
+                    whereBuilder.append(" AND (");
+                    whereBuilder.append(partDesc.getPartitionConditionBuilder().buildDateRangeCondition(partDesc,
+                            flatDesc.getSegment(), segRange,
+                            col -> getTableColumnIdentityQuoted(col, metadata, metaMap, true)));
+                    whereBuilder.append(")");
+                    whereBuilder.append(sep);
+                }
+            }
+        }
+        sql.append(whereBuilder.toString());
+    }
+
+    /**
+     * @return {TABLE_NAME}.{COLUMN_NAME}
+     */
+    private static String getTableColumnIdentityQuoted(TblColRef col, IJdbcMetadata metadata,
+            Map<String, String> metaMap, boolean needQuote) {
+        String tblName = getTableIdentityQuoted(col.getTableRef(), metaMap, metadata, needQuote);
+        String colName = getColumnIdentityQuoted(col, metadata, metaMap, needQuote);
+        return tblName + "." + colName;
+    }
+
+    /**
+     * @return {SCHEMA_NAME}
+     */
+    static String getSchemaQuoted(Map<String, String> metaMap, String database, IJdbcMetadata metadata,
+            boolean needQuote) {
+        String databaseName = fetchValue(database, null, null, metaMap);
+        if (needQuote) {
+            return quoteIdentifier(databaseName, metadata.getDialect());
+        } else {
+            return databaseName;
+        }
+    }
+
+    /**
+     * @return {TABLE_NAME}
+     */
+    static String getTableIdentityQuoted(TableRef tableRef, Map<String, String> metaMap, IJdbcMetadata metadata,
+            boolean needQuote) {
+        String value = fetchValue(tableRef.getTableDesc().getDatabase(), tableRef.getTableDesc().getName(), null,
+                metaMap);
+        String[] res = value.split("\\.");
+        value = res[res.length - 1];
+        if (needQuote) {
+            return quoteIdentifier(value, metadata.getDialect());
+        } else {
+            return value;
+        }
+    }
+
+    /**
+     * @return {TABLE_NAME}
+     */
+    static String getTableIdentityQuoted(String database, String table, Map<String, String> metaMap,
+            IJdbcMetadata metadata, boolean needQuote) {
+        String value = fetchValue(database, table, null, metaMap);
+        String[] res = value.split("\\.");
```

Comment:
looks like there is possible that "value" does not contains any dot

Reply:
If not contains any dot, res will be a string array with one element, it make no mistake.